Madness/Map.swift

```diff
@@ -24,3 +42,11 @@ infix operator >>- {
 	associativity left
 	precedence 150
 }
+
+/// Map operator.
+infix operator <^> {
+	associativity left
+
+	// Higher precedence than `>>-`.
+	precedence 155
```

Comment:
:ok_hand:

I don’t think we need to hold up this PR on https://github.com/thoughtbot/Runes/pull/30 but it’s good to have the conversation started.